Tutorials/i2c.md

```diff
@@ -0,0 +1,110 @@
+  The next 4 bits are stored in their respective OUT_LSB registers. The remaining 4 bits are occupied
+  by 0s. These lower 4 bits (bits 0, 1, 2, and 3 of the OUT_LSB registers) are redundant bits which are not required. The OUT_LSB and OUT_MSB store the 2's complement form of the coordinates.
+
+  The organization of the registers can be seen in the datasheet inside section 6.1 (Data Registers), page number - 21
+  The register address for OUT_X_MSB is 0x01. This can be found at Page 19 of https://www.nxp.com/docs/en/data-sheet/MMA8452Q.pdf
+
+*/
+  i2c.transfer(new Buffer([0x01]), 6, function (error, dataReceived) {
+
+      // This is an exception. If an error is generated, it will throw the error
+      if (error) throw error;
+
+      //This array is going to store the final x,y,z values
+      var out=[];
+
+      // The for loop is iterated 3 times, in order to extract the x,y, and z acceleration values.
+      for (var i=0;i<3;i++){
+      
+      /*
+    
+       We have 16 bits to put together - the most significant 8 bits and the least significant 8 bits so that we can interpret them as one number. This combined value is placed in gCount variable using the OR operation.
+      The gCount variable is a bitwise OR operation between two binary numbers:
+        1. The 8 bits in OUT_MSB for a given coordinate are shifted left by 8 bits in order to make space for the remaining 8 bits
+        of the OUT_LSB.
+        2. The OUT_LSB of the respective coordinate.
+        
+     */
+        var gCount=(dataReceived[i*2] << 8) | dataReceived[(i*2)+1];
+        
+      // The gCount number of the respective coordinate is right shifted by 4 to get rid of the unused lower 0 bits which are 4 in number yielding the 12 bits that the datasheet specifies each coordinate has on page number 21.  
+        gCount=gCount >> 4;
+
+      /*
+
+      The x,y, and z accelerometer values are stored in their respective OUT_MSB and OUT_LSB registers as negative values i.e. their 2's       complement form. 
+      Therefore in order to obtain the correct values, check whether the most significant bit of the OUT_MSB is 1 or 0 i.e whether the         coordinate value is negative or positive. 
+      If it is negative (checked using 0x7F, which is the maximum possible number that can be made from 7 bits), the if condition             changes it to a 2's complement form, thus making it positive and adding a "-" sign in front of it.
+      127 is checking whether we have a 0 or a 1 at the first position - basically its sign.
+
+      */
+        if (dataReceived[i*2] > 0x7F) {
+            gCount = -(1 + 0xFFF - gCount); // Transform into negative 2's complement
+          }
+          
+      // Lastly, normalize the coordinate to get a value between 0 and 1, dividing gCount by 2^10.
+          out[i] = gCount / ((1<<12)/(2*2));
+      }
+      console.log('The x, y, z values are :',out);
+
+  });
+```
```

Comment:
Does your code actually work & respond appropriately when run? I think you're missing a crucial step– look up SYSMOD in the datasheet